Add release notes for 5.21.0 (#2503)
sensu/sensu-docs: content/sensu-go/5.21/release-notes.md

---

## 5.21.0 release notes

> **June 10, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.

Author: ```suggestion
**June 11, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.
```

> **June 11, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.

Author: ```suggestion
**June 15, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.
```

---

## 5.21.0 release notes

**June 9, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.

**RELEASE SYNOPSIS NEEDED HERE**

See the [upgrade guide][1] to upgrade Sensu to version 5.21.0.

**NEW FEATURES:**

> - ([Commercial feature][158] Added [entity count and limit][156] for each entity class in the tabular title in the response for `sensuctl license info` (in addition to the total entity count and limit).

Author: ```suggestion
- ([Commercial feature][158]) Added [entity count and limit][156] for each entity class in the tabular title in the response for `sensuctl license info` (in addition to the total entity count and limit).
```

> - Added `sensuctl user hash-password` command to generate password hashes. The `user` resource now includes a `password_hash` field to specify the password hash instead of the cleartext password.

Author: ```suggestion
- ([Commercial feature][158]) Added OpenSSL-linked binaries for the Sensu agent and backend, with accompanying `--require-openssl` and `--require-fips` flags for the agent and backend.
- Added `sensuctl user hash-password` command to generate password hashes. The `user` resource now includes a `password_hash` field to specify the password hash instead of the cleartext password.
```

- Added the ability to reset passwords via the backend API and `sensuctl user reset-password`.

> - Added support for the AIX PPC64 platform.

Author: ```suggestion
- Added support for the [AIX PPC64][159] platform.
```

> - Added support for the [AIX PPC64][159] platform.

Author: ```suggestion
- Added binary distribution for the Sensu agent for the [AIX PPC64][159] platform.
```

**IMPROVEMENTS:**

- Changed the [default log level for `sensu-agent`][157] to `info` (instead of `warn`).

> - Updated the gostatsd version to 20.3.2. This will eliminate warning messages about ports when shutting down the agent.

Author: ```suggestion
```

---

## 5.21.0 release notes

> **June 9, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.

Author: ```suggestion
**June 10, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.
```

**RELEASE SYNOPSIS NEEDED HERE**

See the [upgrade guide][1] to upgrade Sensu to version 5.21.0.

**NEW FEATURES:**

- ([Commercial feature][158]) Added [entity count and limit][156] for each entity class in the tabular title in the response for `sensuctl license info` (in addition to the total entity count and limit).

> - ([Commercial feature][158]) Added OpenSSL-linked binaries for the Sensu agent and backend, with accompanying `--require-openssl` and `--require-fips` flags for the agent and backend.

Author: ```suggestion
- ([Commercial feature][158]) Added [Linux amd64 OpenSSL-linked binaries][160] for the Sensu agent and backend, with accompanying `--require-openssl` and `--require-fips` flags for the agent and backend.
```

> - ([Commercial feature][158]) Added [Linux amd64 OpenSSL-linked binaries][160] for the Sensu agent and backend, with accompanying `--require-openssl` and `--require-fips` flags for the agent and backend.

Author: ```suggestion
- ([Commercial feature][158]) Added [Linux amd64 OpenSSL-linked binaries][160] for the Sensu agent and backend, with accompanying `--require-fips` and `--require-openssl` flags for the [agent][161] and [backend][162].
```

> - Added `sensuctl user hash-password` command to generate password hashes. The `user` resource now includes a `password_hash` field to specify the password hash instead of the cleartext password.

Author: ```suggestion
- Added `sensuctl user hash-password` command to generate password hashes.
```

- Added the ability to reset passwords via the backend API and `sensuctl user reset-password`.

> - Added binary distribution for the Sensu agent for the [AIX PPC64][159] platform.

Reviewer: We won't be releasing AIX builds for 5.21 as we ran into some technical difficulties.

> - Added binary distribution for the Sensu agent for the [AIX PPC64][159] platform.

Author: ```suggestion
```

**IMPROVEMENTS:**

- Changed the [default log level for `sensu-agent`][157] to `info` (instead of `warn`).

> - Updated the gostatsd version to 20.3.2.

Author: ```suggestion
- Updated the gostatsd version to 20.3.2. This will eliminate warning messages about ports when shutting down the agent.
```

> - Updated the gostatsd version to 20.3.2. This will eliminate warning messages about ports when shutting down the agent.

Reviewer: We will need to revisit this line closer to the release as we ran into some issues with gostatsd.

Reviewer: We'll be reverting this change, so this line should be removed.

---

## 5.21.0 release notes

**June 15, 2020** &mdash; The latest release of Sensu Go, version 5.21.0, is now available for download.

> **RELEASE SYNOPSIS NEEDED HERE**

Author: ```suggestion
The latest release of Sensu Go, version 5.21.0, is now available for download! This release delivers several enhancements and fixes. The most significant enhancements involve user management: you can now back up users via `sensuctl dump`, restore users via `sensuctl create`, and reset user passwords via the backend API. We also tuned Sensu Go agent logging and changed the default log level from warning to info. Plus, we crushed a number of nasty bugs: checks configured with missing hooks can no longer crash the agent, proxy check request errors do not block scheduling for other entities, and more!
```

[153]: /sensu-go/5.20/dashboard/overview/
[154]: /sensu-go/5.20/sensuctl/reference/#extend-sensuctl-with-commands
[155]: /sensu-go/5.20/reference/agent/#discover-processes
[156]: /sensu-go/5.21/reference/license/#view-entity-count-and-entity-limit
[157]: /sensu-go/5.21/reference/agent/#log-level

> [158]: /sensu-go/5.21/commercial/

Author: ```suggestion
[158]: /sensu-go/5.21/commercial/
[159]: /sensu-go/5.21/platforms/#aix
```

[153]: /sensu-go/5.20/dashboard/overview/
[154]: /sensu-go/5.20/sensuctl/reference/#extend-sensuctl-with-commands
[155]: /sensu-go/5.20/reference/agent/#discover-processes
[156]: /sensu-go/5.21/reference/license/#view-entity-count-and-entity-limit
[157]: /sensu-go/5.21/reference/agent/#log-level
[158]: /sensu-go/5.21/commercial/

> [159]: /sensu-go/5.21/platforms/#aix

Author: ```suggestion
```

[153]: /sensu-go/5.20/dashboard/overview/
[154]: /sensu-go/5.20/sensuctl/reference/#extend-sensuctl-with-commands
[155]: /sensu-go/5.20/reference/agent/#discover-processes
[156]: /sensu-go/5.21/reference/license/#view-entity-count-and-entity-limit
[157]: /sensu-go/5.21/reference/agent/#log-level
[158]: /sensu-go/5.21/commercial/

> [159]: /sensu-go/5.21/platforms/#aix

Author: ```suggestion
[159]: /sensu-go/5.21/platforms/#aix
[160]: /sensu-go/5.21/platforms/#linux
```

[153]: /sensu-go/5.20/dashboard/overview/
[154]: /sensu-go/5.20/sensuctl/reference/#extend-sensuctl-with-commands
[155]: /sensu-go/5.20/reference/agent/#discover-processes
[156]: /sensu-go/5.21/reference/license/#view-entity-count-and-entity-limit
[157]: /sensu-go/5.21/reference/agent/#log-level
[158]: /sensu-go/5.21/commercial/
[159]: /sensu-go/5.21/platforms/#aix

> [160]: /sensu-go/5.21/platforms/#linux

Author: ```suggestion
[160]: /sensu-go/5.21/platforms/#linux
[161]: /sensu-go/5.21/reference/agent#fips-openssl
[162]: /sensu-go/5.21/reference/backend#fips-openssl
```
